virtual_ecosystem/models/animals/animal_cohorts.py

```diff
@@ -748,3 +765,49 @@ def migrate_juvenile_probability(self) -> float:
         probability_of_dispersal = velocity / grid_side
 
         return min(1.0, probability_of_dispersal)
+
+    def inflict_non_predation_mortality(
+        self, dt: float, carcass_pool: CarcassPool
+    ) -> None:
+        """Inflict combined background, senescence, and starvation mortalities.
+
+        Args:
+            dt: The time passed in the timestep (days).
+            carcass_pool: The local carcass pool to which dead individuals go.
+
+        """
+
+        pop_size = self.individuals
+        mass_current = self.mass_current
+
+        t_to_maturity = self.time_to_maturity
+        t_since_maturity = self.time_since_maturity
+        mass_max = self.functional_group.adult_mass  # this might not be only solution
+
+        u_bg = sf.background_mortality(
+            self.constants.u_bg
+        )  # constant background mortality
+
+        if self.is_mature is True:
+            # senescence mortality is only experienced by mature adults.
+            u_se = sf.senescence_mortality(
+                self.constants.lambda_se, t_to_maturity, t_since_maturity
+            )  # senesence mortality
+        elif self.is_mature is False:
+            u_se = 0.0
```

Comment:
This can be slightly re-organised to save a conditional:
```suggestion
        u_se = 0.0
        if self.is_mature:
            # senescence mortality is only experienced by mature adults.
            u_se = sf.senescence_mortality(
                self.constants.lambda_se, t_to_maturity, t_since_maturity
            )  # senesence mortality
```